src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static partial class ISymbolExtensions
+    {
+        /// <summary>
+        /// Checks if 'symbol' is accessible from within 'within', with
```

Comment:
>Checks if 'symbol' is accessible from within 'within', with [](start = 12, length = 59)

The doc comment sounds so innocent and doesn't provide any clues about inaccuracies built into this component and how they can affect the result. I think that at least all possibilities to get false positive and false negative results should be clearly enumerated with specific examples, etc. So that potential consumers clearly understand what they are getting into.

Reply:
I'm not aware of any false positive or false negative scenarios. Please identify *specific* scenarios that you theorize could be incorrect.

Reply:
> I'm not aware of any false positive or false negative scenarios. 

You are not aware or are you claiming there are not any?

---
In reply to: [185302839](https://github.com/dotnet/roslyn/pull/26367#discussion_r185302839) [](ancestors = 185302839)

Reply:
The code now uses metadata name to compare names, and .Equals to compare assemblies.

Please identify *specific* scenarios that you theorize could be incorrect, or else stop making this claim.


---
In reply to: [185303824](https://github.com/dotnet/roslyn/pull/26367#discussion_r185303824) [](ancestors = 185303824,185302839)